### ISSUE TITLE
Error: connect ECONNREFUSED - wemo - port problems?

### ISSUE THREAD

**Issue by Achronite** [node-red/node-red#523]
I have a problem with node-red-node-wemo interacting with a wemo switch using the wemo in node.  Intermittently the wemo nodes in node-red report 'Error:  connect ECONNREFUSED'.  I have performed some research and it appears that the port the wemo uses can vary between 49152 and 49155, so I suspect this may be the problem.  Is there any way the node-red-node-wemo driver can be changed to automatically discover and use the correct port?

For example something like this shell script snippit from http://wemo.forumatic.com/viewtopic.php?f=2&t=5 :

    for PTEST in 49152 49153 49154 49155
    do
            PORTTEST=$(curl -s -m 3 $IP:$PTEST | grep "404")
                     
            if [ "$PORTTEST" != "" ]
               then
               PORT=$PTEST
               break
            fi
    done

I have run the full script on that page and it says my wemo is currently on port 49153 - and I am getting the ECONNREFUSED error in node-red.

EDIT: I have just found out that if i change the default port number in the file 'node_modules/node-red-node-wemo/node_modules/wemo/wemo.js' from 49154 to 49153 it fixes the issue - confirming that the problem is because the wemo does not always operate on port 49154.

**Comment by dceejay** [comment]
Yes - seems to be a "known problem" - tends to be one port and reverts to it on reset - but can also move... we can easily make it configurable, or auto detection slightly harder. (One is a 5 min fix, the other will need a bit more thinking about). Happy to do either / or / both.

**Comment by hardillb** [comment]
I have some code that looks up the port number every time you need to send a command, it's in my prototype code for the Wemo lightbulbs (https://gist.github.com/hardillb/1279241bb886ee28c05b). I'm planning to try and integrate it with one of the Wemo nodes soon, so we can hopefully use this in the Node-RED node

**Comment by knolleary** [comment]
The user is unlikely to know what the port is, so allowing the set it isn't going to be that helpful. It should just work.

**Comment by Achronite** [comment]
I've implemented the workaround by adding the port as a configuration parameter to the wemo in and wemo out nodes (locally only - never checked anything in to github before!). But I think discovery is the best option by using the 'Friendly Name' instead of IP & Port.

I did try to call the 'WeMo.Search' function from node-wemo.js (as this seems to have the facility) - but node-red-node-wemo doesn't seem to be able to call it - I'm pretty new to javascript programming, so it's more than likely ignorance on my part!
